src/Compilers/CSharp/Portable/Symbols/Source/LocalFunctionSymbol.cs

```diff
@@ -503,9 +503,9 @@ public sealed override bool Equals(Symbol symbol, TypeCompareKind compareKind)
         {
             if ((object)this == symbol) return true;
 
-            var localFunction = symbol as LocalFunctionSymbol;
-            return (object)localFunction != null
-                && localFunction._syntax == _syntax;
+            return symbol is LocalFunctionSymbol localFunction
```

Comment:
>return symbol is LocalFunctionSymbol localFunction [](start = 12, length = 50)

It is not clear why this code is changed. Consider reverting the file to its original state.